File: week11/2/Prob2.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        vector<int> climbcost;
        int N = cost.size();
        
        climbcost.push_back(cost[0]);
        climbcost.push_back(cost[1]);
        if(N < 3){
            return *min_element(climbcost.begin(), climbcost.end());
        }else{
            int i = 2;
            while(N - 1 > i){
                climbcost.push_back(min(climbcost[i-1], climbcost[i-2])+cost[i]);
                i++;

            }
            
            int nn = climbcost.size();
            
            return min(climbcost[nn-1], climbcost[nn-2]+cost[N-1]);

        }
    }
};

int main(){
    Solution sol;
    vector<int> cost;
    cost.push_back(10);
    cost.push_back(15);
    cost.push_back(20);
    sol.minCostClimbingStairs(cost);
}

File: week11/2/Prob2.py
from typing import List

class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        climbcost = []
        N = len(cost)
        # base case
        climbcost.append(cost[0])
        climbcost.append(cost[1])
        if N < 3:
            return min(climbcost)
        else:

            i = 2
            while N-1 > i:
                climbcost.append(min(climbcost[i-1], climbcost[i-2])+cost[i])
                i += 1

            return min(climbcost[-1], climbcost[-2]+cost[-1])


